src/components/ChangeDataForm.vue

```diff
@@ -0,0 +1,262 @@
+				</template>
+				<template #content>
+					<InputMask v-model="patientData.address.zipCode" mask="99-999" />
+				</template>
+			</Inplace>
+			<Button class="" label="Confirm" @click="compareData" :loading="isLoading" />
+		</template>
+	</Card>
+</template>
+
+<script setup>
+import { useUserSession } from "@/services/useUserSession";
+import Inplace from "primevue/inplace";
+import InputText from "primevue/inputtext";
+import Card from "primevue/card";
+import InputMask from "primevue/inputmask";
+import Button from "primevue/button";
+import { changeData } from "@/services/api";
+import { useToast } from "primevue/usetoast";
+import Password from "primevue/password";
+import Divider from "primevue/divider";
+import Cookies from "js-cookie";
+import { ref, defineProps, defineEmits } from "vue";
+
+defineProps({
+	visible: {
+		type: Boolean,
+		default: false,
+	},
+});
+const emit = defineEmits(["hide"]);
+
+const isLoading = ref(false);
+const toast = useToast();
+
+const compare = (obj1, obj2, properties = {}, objKey = null) => {
+	Object.keys(obj1).forEach((key) => {
+		if (typeof obj1[key] === "object") {
+			compare(obj1[key], obj2[key], properties, key);
+		} else if (obj1[key] != obj2[key]) {
+			if (objKey) {
+				if (!properties[objKey]) properties[objKey] = {};
+				properties[objKey][key] = obj1[key];
+			} else properties[key] = obj1[key];
+		}
+	});
+	return properties;
+};
+
+const compareData = () => {
+	const changes = compare(patientData, userData);
+	if (Object.keys(changes).length === 0) {
+		toast.add({
+			severity: "error",
+			summary: "Error",
+			detail: "No changes made",
+			life: 3000,
+		});
+	} else {
+		console.log(changes);
+		sendData(changes);
+	}
+};
+
+const sendData = (changes) => {
+	isLoading.value = true;
+	changeData(changes)
+		.then((response) => {
+			useUserSession().userInfo = response;
+			Cookies.set("user-info", JSON.stringify(response));
+            emit("hide");
+            toast.add({
+                severity: "success",
+                summary: "Success",
+                detail: "Successfully changed your data",
+                life: 3000,
+            });
+		})
+		.catch((err) => {
+			console.error(err);
+			toast.add({
+				severity: "error",
+				summary: err?.response?.statusText || "Error",
+				detail: err?.response?.data?.msg || "Could not process your request",
+				life: 3000,
+			});
+		})
+		.finally(() => {
+			isLoading.value = false;
+		});
+};
+
+const userData = JSON.parse(JSON.stringify(useUserSession().userInfo));
+userData["password"] = "";
+
+const patientData = {
+	firstName: userData.firstName,
+	lastName: userData.lastName,
+	password: "",
+	address: {
+		city: userData.address.city,
+		zipCode: userData.address.zipCode,
+		street: userData.address.street,
+		houseNumber: userData.address.houseNumber,
+		localNumber: userData.address.localNumber,
+	},
+};
+</script>
+
+<style lang="scss">
```

Comment:
W 99% przypadkach staramy się nie usuwać "scoped" ze stylu, bo to grozi wyciekiem stylów z komponentu do całej aplikacji. Jeśli już tak robimy, to musimy pododawać klasy typu "moj-komponent-card". Jeśli jest problem z modyfikacją zagnieżdżonych stylów, to wykorzystujemy `::v-deep`